internal/datasource/config/config_writer.go

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/nginx/agent/v3/api/grpc/instances"
+	"github.com/nginx/agent/v3/internal/client"
+	"github.com/nginx/agent/v3/internal/datasource/nginx"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6@v6.7.0 -generate
+//counterfeiter:generate -o mock_config_writer.go . ConfigWriterInterface
+//go:generate sh -c "grep -v github.com/nginx/agent/v3/internal/datasource/config mock_config_writer.go | sed -e s\\/config\\\\.\\/\\/g > mock_config_writer_fixed.go"
+//go:generate mv mock_config_writer_fixed.go mock_config_writer.go
```

Comment:
Probably should do this the same way as Donal is in his PR: https://github.com/nginx/agent/pull/547

Reply:
I think I might need the other changes Donal made for that to work so might wait till I that is merged in. Think I need the change from `mock` to `fake` could be wrong though